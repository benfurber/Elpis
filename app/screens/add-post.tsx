// import { NavigationType, Post } from "interfaces";
// import React, { Component } from "react";
// import { TextInput, TouchableOpacity, ScrollView } from "react-native-gesture-handler";
// import { Analytics, firstSentence } from "utils";
// import { labels } from "labels";
// import { ADD_COMMENT, ADD_REPLY } from "mutations";
// import { View, Text, StyleSheet } from "react-native";
// import { Mutation } from "react-apollo";
// import { BackgroundContainer } from "components";
// import { Icon } from "react-native-vector-icons/Icon";
// import { Title } from "app/components/notifications/title";
// import { colours, layout, typography } from "styles";
// import { withMappedNavigationParams } from "react-navigation-props-mapper";

// interface Props {
//   navigation: NavigationType;
//   postId: Post["id"];
// }

// interface State {
//   content: string;
//   imageUrl: string;
//   title: string;
// }

// class AddPostScreen extends Component<Props, State> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       content: "",
//       imageUrl: "",
//       title: "",
//     };
//   }

//   componentDidMount() {
//     // Analytics.trackContent({
//     //   contentType: "AddPost",
//     //   contentId: this.props.postId,
//     // });
//   }

//   componentDidUpdate(_props, state) {
//   }

//   onSubmitEditing(query, id) {
//     const { content, title, imageUrl } = this.state;

//     query({
//       variables: {
//         content,
//         id,
//       },
//     }).then(() => {
//       this.setState({ content: "", textInputEditable: true });
//       this.props.navigation.pop();
//     });
//   }

//   input(mutation) {

//     const args = {
//       autoFocus: true,
//       editable: true,
//       multiline: true,
//       placeholder: labels.addPlaceholderTitle,
//       onChangeText: content => this.setState({ content }),
//       value: this.state.content,
//       returnKeyLabel: labels.submit,
//     };

//     if (mutation === ADD_COMMENT) {
//       return (createComment, {}) => (
//         <View style={styles.textInputContainer}>
//           <Text style={styles.label}>{labels.title}</Text>
//           <TextInput
//             onSubmitEditing={() =>
//               this.onSubmitEditing(createComment, this.props.postId)
//             }
//             style={styles.title}
//             {...args}
//           />
//           <Text style={styles.label}>{labels.body}</Text>
//           <TextInput
//             {...args}
//             autoFocus={false}
//             ref={input => (this.secondTextInput = input)}
//             onSubmitEditing={() =>
//               this.onSubmitEditing(createComment, this.props.postId)
//             }
//             onChangeText={textInputExtra => this.setState({ textInputExtra })}
//             value={this.state.textInputExtra}
//             style={styles.text}
//             placeholder={labels.addPlaceholderBody}
//           />
//         </View>
//       );
//     }

//     return (createPost: any, {}: any) => (
//       <TextInput
//         onSubmitEditing={() =>
//           this.onSubmitEditing(createPost, this.props.postId)
//         }
//         style={styles.text}
//         {...args}
//       />
//     );
//   }

//   renderAddResponse(mutation) {
//     return <Mutation mutation={mutation}>{this.input(mutation)}</Mutation>;
//   }

//   render() {
//     const { commentId } = this.props;

//     const text = commentId ? labels.addYourReply : labels.addNewTopic;
//     const mutation = commentId ? ADD_REPLY : ADD_COMMENT;

//     return (
//       <BackgroundContainer>
//         <View style={styles.header}>
//           <View style={styles.closeContainer}>
//             <TouchableOpacity onPress={() => this.props.navigation.pop()}>
//               <Icon name="times-circle" size={30} />
//             </TouchableOpacity>
//           </View>
//           <Title style={styles.ctaText} text={text + ":"} small />
//         </View>
//         <View style={styles.body}>
//           <ScrollView>
//             <View style={styles.row}>{this.renderAddResponse(mutation)}</View>
//           </ScrollView>
//         </View>
//       </BackgroundContainer>
//     );
//   }
// }

// const styles = StyleSheet.create({
//   body: {
//     backgroundColor: colours.whiteTransparent,
//     borderTopLeftRadius: layout.borderRadiusL,
//     borderTopRightRadius: layout.borderRadiusL,
//     flex: 1,
//     paddingTop: layout.spacing,
//   },
//   closeContainer: {
//     flexDirection: "row-reverse",
//     padding: layout.spacing,
//   },
//   ctaText: {
//     paddingHorizontal: layout.spacing,
//   },
//   header: {},
//   label: {
//     color: colours.mediumGrey,
//     fontWeight: "bold",
//     paddingTop: layout.spacingXL,
//     paddingHorizontal: layout.spacingL,
//   },
//   row: {
//     alignItems: "baseline",
//     flexDirection: "row",
//   },
//   textInputContainer: {
//     width: "100%",
//     flexDirection: "column",
//   },
//   title: {
//     padding: layout.spacingL,
//     fontFamily: "creteround-regular",
//     fontSize: typography.fontSizeXL,
//     flexWrap: "wrap",
//   },
//   text: {
//     fontFamily: "creteround-regular",
//     fontSize: typography.fontSizeL,
//     padding: layout.spacingL,
//     width: "100%",
//   },
// });

// const wrappedAddPostScreen = withMappedNavigationParams()(AddPostScreen);
// export {
//   wrappedAddPostScreen as AddPostScreen,
//   AddPostScreen as UnwrappedAddPostScreen,
// };