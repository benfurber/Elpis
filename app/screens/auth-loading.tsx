import React, { Component } from "react";
import { StyleSheet, View } from "react-native";
import AsyncStorage from "@react-native-community/async-storage";

import { BackgroundContainer, Loading } from "components";
import { NavigationType } from "interfaces";
import { USER_DETAILS } from "queries";
import { Analytics, client } from "utils";

interface Props {
  navigation: NavigationType;
}

class AuthLoadingScreen extends Component<Props> {
  async componentDidMount() {
    const { navigate } = this.props.navigation;
    const token = await AsyncStorage.getItem("token");

    if (token === null) {
      return navigate("Login");
    }

    this.callClient();
  }

  async callClient() {
    const { navigate } = this.props.navigation;

    const { data } = await client.query({
      fetchPolicy: "network-only",
      query: USER_DETAILS,
    });
    Analytics.identifyUser(data.me.id);

    return navigate(data.me.onboarded ? "Main" : "Onboarding");
  }

  render() {
    return (
      <BackgroundContainer>
        <View style={styles.container}>
          <Loading />
        </View>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    alignSelf: "stretch",
    justifyContent: "center",
  },
});

export { AuthLoadingScreen };
