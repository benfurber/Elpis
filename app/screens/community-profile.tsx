import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

import { withMappedNavigationParams } from "react-navigation-props-mapper";

import {
  BackgroundContainer,
  ButtonClose,
  CommunityDetails,
  Loading,
} from "components";
import { Community, NavigationType } from "interfaces";
import { COMMUNITY } from "queries";
import { colours, layout } from "styles";
import { Analytics, client } from "utils";

interface Props {
  navigation: NavigationType;
  communityId: Community["id"];
}

interface State {
  community: null | Community;
  isLoading: boolean;
}

class CommunityProfileScreen extends Component<Props, State> {
  state = {
    community: null,
    isLoading: true,
  };

  async fetchUser() {
    const { communityId } = this.props;

    const query = COMMUNITY;
    const variables = { id: communityId };

    const result = await client.query({ query, variables });
    this.setState({ community: result.data.community, isLoading: false });
  }

  componentDidMount() {
    Analytics.track("Community Profile");
    this.fetchUser();
  }

  render() {
    const { navigation } = this.props;
    const { isLoading, community } = this.state;

    return (
      <BackgroundContainer header={<ButtonClose navigation={navigation} />}>
        <View style={styles.container}>
          {isLoading && <Loading blueMode />}
          {community && (
            <CommunityDetails navigation={navigation} community={community} />
          )}
        </View>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: colours.whiteTransparent,
    borderTopLeftRadius: layout.borderRadiusL,
    borderTopRightRadius: layout.borderRadiusL,
    height: "100%",
    padding: layout.spacingL,
  },
});

const wrappedCommunityProfileScreen = withMappedNavigationParams()(
  CommunityProfileScreen,
);
export {
  wrappedCommunityProfileScreen as CommunityProfileScreen,
  CommunityProfileScreen as UnwrappedCommunityProfileScreen,
};
