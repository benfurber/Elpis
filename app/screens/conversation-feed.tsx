import React, { Component } from "react";
import { StyleSheet } from "react-native";

import { BackgroundContainer, ConversationList, Title } from "components";
import { NavigationType } from "interfaces";
import { labels } from "labels";
import { layout, typography } from "styles";
import { Analytics } from "utils";

interface Props {
  navigation: NavigationType;
}

class ConversationFeedScreen extends Component<Props> {
  componentDidMount() {
    Analytics.track("Conversation Feed");
  }

  render() {
    const { navigation } = this.props;

    return (
      <BackgroundContainer>
        <Title style={styles.title} text={labels.messages.title} />
        <ConversationList navigation={navigation} />
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: typography.fontSizeXL,
    margin: layout.spacingL,
  },
});

export { ConversationFeedScreen };
