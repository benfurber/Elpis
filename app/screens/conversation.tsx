import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

import { withMappedNavigationParams } from "react-navigation-props-mapper";

import {
  BackButton,
  BackgroundContainer,
  ConversationScreenHeader,
  ConversationQuery,
  KeyboardAddMessage,
} from "components";
import { Conversation, NavigationType, User } from "interfaces";
import { layout, colours } from "styles";
import { labels } from "labels";
import { Analytics } from "utils";

interface Props {
  backToText?: string;
  currentUserId: string;
  id: Conversation["id"];
  remainingParticipants: User[];
  navigation: NavigationType;
}

interface State {
  feedScrollView: any;
}

class ConversationScreen extends Component<Props, State> {
  state = {
    feedScrollView: null,
  };

  componentDidMount() {
    Analytics.track("Conversation");
  }

  render() {
    const {
      backToText,
      currentUserId,
      navigation,
      remainingParticipants,
      id,
    } = this.props;

    const { feedScrollView } = this.state;

    return (
      <BackgroundContainer>
        <BackButton
          navigation={navigation}
          onPress={() => navigation.popToTop()}
          text={backToText || labels.back.toConversations}
        />
        <ConversationScreenHeader
          navigation={navigation}
          remainingParticipants={remainingParticipants}
        />
        <View style={styles.container}>
          <ConversationQuery
            id={id}
            currentUserId={currentUserId}
            navigation={navigation}
            setFeedScrollView={feedScrollView => this.setState(feedScrollView)}
          />
        </View>
        <KeyboardAddMessage
          conversationId={id}
          feedScrollView={feedScrollView}
        />
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "stretch",
    backgroundColor: colours.whiteTransparent,
    flex: 1,
    paddingTop: layout.spacing,
    width: "100%",
  },
});

const wrappedConversationScreen = withMappedNavigationParams()(
  ConversationScreen,
);
export {
  wrappedConversationScreen as ConversationScreen,
  ConversationScreen as UnwrappedConversationScreen,
};
