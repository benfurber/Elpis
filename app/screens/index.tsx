export { AddReplyScreen } from "./add-reply";
export { AddTopicScreen } from "./add-topic";
export { AuthLoadingScreen } from "./auth-loading";
export { FeedScreen } from "./feed";
export { FeedbackScreen } from "./feedback";
export { FormContainerScreen } from "./form-container";
export { ImageBrowserScreen } from "./image-browser";
export { MenuScreen } from "./menu";
export { NotificationScreen } from "./notification";
export { OnboardingAddPasswordScreen } from "./onboarding-add-password";
export { OnboardingFeaturesScreen } from "./onboarding-features";
export { OnboardingFinalScreen } from "./onboarding-final";
export { OnboardingWelcomeScreen } from "./onboarding-welcome";
export { PostScreen } from "./post";
export { StaticContentScreen } from "./static-content";
export { WebBrowserScreen } from "./web-browser";
export { WelcomeScreen } from "./welcome";
