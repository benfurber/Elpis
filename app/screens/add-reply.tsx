import React, { Component } from "react";
import { StyleSheet, TextInput } from "react-native";

import { Mutation } from "react-apollo";
import { withMappedNavigationParams } from "react-navigation-props-mapper";

import { Comment, NavigationType } from "interfaces";
import { labels } from "labels";
import { ADD_REPLY } from "mutations";
import { FormContainerScreen } from "screens";
import { layout, typography, colours } from "styles";

interface Props {
  commentId: Comment["id"];
  navigation: NavigationType;
}

interface State {
  textInput: string;
  textInputEditable: boolean;
}

class AddReplyScreen extends Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      textInputEditable: true,
      textInput: "",
    };
  }

  onSubmitEditing(query) {
    const { textInput } = this.state;
    const { commentId, navigation } = this.props;

    this.setState({ textInputEditable: false });

    query({
      variables: {
        content: textInput,
        id: commentId,
      },
    }).then(() => {
      this.setState({ textInput: "", textInputEditable: true });
      navigation.dismiss();
    });
  }

  renderAddResponse(mutation) {
    return (
      <Mutation mutation={mutation}>
        {(createReply, {}) => (
          <TextInput
            onSubmitEditing={() => this.onSubmitEditing(createReply)}
            style={styles.text}
            autoFocus={true}
            editable={this.state.textInputEditable}
            multiline={true}
            placeholder={labels.addPlaceholderTitle}
            onChangeText={textInput => this.setState({ textInput })}
            value={this.state.textInput}
            returnKeyLabel={labels.submit}
          />
        )}
      </Mutation>
    );
  }

  render() {
    const { commentId, navigation } = this.props;

    return (
      <FormContainerScreen
        analyticsContent={{ contentType: "AddReply", contentId: commentId }}
        navigation={navigation}
        title={labels.addYourReply}
      >
        {this.renderAddResponse(ADD_REPLY)}
      </FormContainerScreen>
    );
  }
}

const styles = StyleSheet.create({
  label: {
    color: colours.mediumGrey,
    fontWeight: "bold",
    paddingTop: layout.spacingXL,
    paddingHorizontal: layout.spacingL,
  },
  textInputContainer: {
    width: "100%",
    flexDirection: "column",
  },
  text: {
    fontFamily: "creteround-regular",
    fontSize: typography.fontSizeL,
    padding: layout.spacingL,
    width: "100%",
  },
});

const wrappedAddReplyScreen = withMappedNavigationParams()(AddReplyScreen);
export {
  wrappedAddReplyScreen as AddReplyScreen,
  AddReplyScreen as UnwrappedAddReplyScreen,
};
