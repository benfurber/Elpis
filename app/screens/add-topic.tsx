import React, { Component } from "react";
import { Text, TextInput, View } from "react-native";

import { Mutation } from "react-apollo";
import { withMappedNavigationParams } from "react-navigation-props-mapper";
import { DocumentNode } from "graphql";

import { Comment as CommentType, NavigationType, Post } from "interfaces";
import { labels } from "labels";
import { ADD_COMMENT, UPDATE_COMMENT } from "mutations";
import { FormContainerScreen } from "screens";
import { form } from "styles";
import { firstSentence } from "utils";

interface Props {
  currentComment?: {
    content: CommentType["content"];
    id: CommentType["id"];
    title: CommentType["title"];
  };
  navigation: NavigationType;
  postId: Post["id"];
}

interface State {
  content: CommentType["content"];
  editable: boolean;
  id: string;
  mutation: DocumentNode;
  title: CommentType["title"];
}

class AddTopicScreen extends Component<Props, State> {
  secondField = TextInput as any;

  state = {
    content: "",
    editable: true,
    id: this.props.postId,
    mutation: ADD_COMMENT,
    title: "",
  };

  componentDidMount() {
    const { currentComment } = this.props;
    if (currentComment) {
      const { content, id, title } = currentComment;
      const mutation = UPDATE_COMMENT;

      this.setState({ content, id, mutation, title });
    }
  }

  componentDidUpdate(_props, state) {
    const { title } = state;

    if (title) {
      const title = firstSentence(state.title);

      if (title !== state.title) {
        this.setState({ title });
        return this.secondField.focus();
      }
    }
  }

  onSubmitEditing(call) {
    const { title, id, content } = this.state;
    this.setState({ editable: false });

    call({
      variables: {
        content,
        id,
        title: title && title.length > 0 ? title : null,
      },
    }).then(() => {
      this.props.navigation.dismiss();
    });
  }

  form() {
    const { title, editable, content } = this.state;

    const args = {
      editable,
      multiline: true,
    };

    return (createComment, {}) => (
      <View style={form.fieldContainer}>
        <Text style={form.label}>{labels.title}</Text>
        <TextInput
          autoFocus={false}
          onChangeText={title => this.setState({ title })}
          onSubmitEditing={() => this.secondField.focus()}
          style={form.title}
          placeholder={labels.addPlaceholderTitle}
          value={title}
          returnKeyLabel={labels.next}
          returnKeyType="next"
          {...args}
        />
        <Text style={form.label}>{labels.body} *</Text>
        <TextInput
          {...args}
          autoFocus={true}
          ref={input => (this.secondField = input)}
          onChangeText={content => this.setState({ content })}
          onSubmitEditing={() => this.onSubmitEditing(createComment)}
          value={content}
          style={form.text}
          placeholder={labels.addPlaceholderBody}
          returnKeyLabel={labels.submit}
          returnKeyType="send"
        />
      </View>
    );
  }

  render() {
    const { mutation } = this.state;

    return (
      <FormContainerScreen
        analyticsContent={{
          contentId: this.props.postId,
          contentType: "AddTopic",
        }}
        navigation={this.props.navigation}
        title={labels.addNewTopic}
      >
        <Mutation mutation={mutation}>{this.form()}</Mutation>
      </FormContainerScreen>
    );
  }
}

const wrappedAddTopicScreen = withMappedNavigationParams()(AddTopicScreen);
export {
  wrappedAddTopicScreen as AddTopicScreen,
  AddTopicScreen as UnwrappedAddTopicScreen,
};
