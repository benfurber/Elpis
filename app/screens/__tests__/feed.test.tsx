import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { FeedScreen } from "screens";

describe("FeedScreen", () => {
  it("renders correctly", () => {
    const component = shallow(<FeedScreen />);

    expect(component).toMatchSnapshot();
  });
});
