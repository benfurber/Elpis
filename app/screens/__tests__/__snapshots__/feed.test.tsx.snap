// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedScreen renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flex": 1,
            }
          }
        >
          <Logo />
        </View>,
        <View
          style={
            Object {
              "flex": 6,
            }
          }
        >
          <FlatList
            data={
              Array [
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-17T00:00:00.000Z,
                  "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                  "id": "98y723-23048-23480",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_1.jpg",
                  },
                },
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-11T00:00:00.000Z,
                  "description": null,
                  "id": "324869-saf80as-ssae",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_2.jpg",
                  },
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Logo />,
          "style": Object {
            "alignItems": "stretch",
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FlatList
            data={
              Array [
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-17T00:00:00.000Z,
                  "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                  "id": "98y723-23048-23480",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_1.jpg",
                  },
                },
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-11T00:00:00.000Z,
                  "description": null,
                  "id": "324869-saf80as-ssae",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_2.jpg",
                  },
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
          "style": Object {
            "flex": 6,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "author": Object {
                  "avatarPath": Object {
                    "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                  },
                },
                "date": 2018-02-17T00:00:00.000Z,
                "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                "id": "98y723-23048-23480",
                "imagePath": Object {
                  "testUri": "../../../app/assets/images/image_post_1.jpg",
                },
              },
              Object {
                "author": Object {
                  "avatarPath": Object {
                    "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                  },
                },
                "date": 2018-02-11T00:00:00.000Z,
                "description": null,
                "id": "324869-saf80as-ssae",
                "imagePath": Object {
                  "testUri": "../../../app/assets/images/image_post_2.jpg",
                },
              },
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flex": 1,
              }
            }
          >
            <Logo />
          </View>,
          <View
            style={
              Object {
                "flex": 6,
              }
            }
          >
            <FlatList
              data={
                Array [
                  Object {
                    "author": Object {
                      "avatarPath": Object {
                        "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                      },
                    },
                    "date": 2018-02-17T00:00:00.000Z,
                    "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                    "id": "98y723-23048-23480",
                    "imagePath": Object {
                      "testUri": "../../../app/assets/images/image_post_1.jpg",
                    },
                  },
                  Object {
                    "author": Object {
                      "avatarPath": Object {
                        "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                      },
                    },
                    "date": 2018-02-11T00:00:00.000Z,
                    "description": null,
                    "id": "324869-saf80as-ssae",
                    "imagePath": Object {
                      "testUri": "../../../app/assets/images/image_post_2.jpg",
                    },
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </View>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Logo />,
            "style": Object {
              "alignItems": "stretch",
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FlatList
              data={
                Array [
                  Object {
                    "author": Object {
                      "avatarPath": Object {
                        "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                      },
                    },
                    "date": 2018-02-17T00:00:00.000Z,
                    "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                    "id": "98y723-23048-23480",
                    "imagePath": Object {
                      "testUri": "../../../app/assets/images/image_post_1.jpg",
                    },
                  },
                  Object {
                    "author": Object {
                      "avatarPath": Object {
                        "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                      },
                    },
                    "date": 2018-02-11T00:00:00.000Z,
                    "description": null,
                    "id": "324869-saf80as-ssae",
                    "imagePath": Object {
                      "testUri": "../../../app/assets/images/image_post_2.jpg",
                    },
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
            "style": Object {
              "flex": 6,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-17T00:00:00.000Z,
                  "description": "Maju Coutinho estreia na bancada do Jornal Nacional 

Maria Júlia Coutinho estreou neste sábado, 16 de Fevereiro, como âncora do Jornal Nacional. O debute da primeira mulher negra na bancada do principal jornalístico da televisão brasileira rendeu elogios e muita comoção nas redes sociais.",
                  "id": "98y723-23048-23480",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_1.jpg",
                  },
                },
                Object {
                  "author": Object {
                    "avatarPath": Object {
                      "testUri": "../../../app/assets/images/empower_two_women_logo.png",
                    },
                  },
                  "date": 2018-02-11T00:00:00.000Z,
                  "description": null,
                  "id": "324869-saf80as-ssae",
                  "imagePath": Object {
                    "testUri": "../../../app/assets/images/image_post_2.jpg",
                  },
                },
              ],
              "disableVirtualization": false,
              "horizontal": false,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
